tests/test_gzip_request.js

```diff
@@ -36,6 +36,35 @@ it('should accept and decode gzip encoded application/json', done => {
   req.end()
 })
 
+it('should accept and decode gzip encoded application/json, when headers come node-fetch style as array', done => {
```

Comment:
```suggestion
it('should accept and decode gzip encoded application/json, when headers come from a client as an array', done => {
```
We tend to try to keep most of the tests client lib agnostic.